lib/interfaces/uncaught.js

```diff
@@ -86,13 +38,30 @@ function attachUncaughtExceptionHandler(client, config) {
  *  exceptions, otherwise return the process instance
  */
 function handlerSetup(client, config) {
+  /**
+   * The actual exception handler creates a new instance of `ErrorMessage`,
+   * extracts infomation from the propagated `Error` and marshals it into the
+   * `ErrorMessage` instance, attempts to send this `ErrorMessage` instance to
+   * the Stackdriver Error Reporting API. Subsequently the process is
+   * terminated.
+   * @function uncaughtExceptionHandler
+   * @listens module:process~event:uncaughtException
+   * @param {Error} err - The error that has been uncaught to this point
+   * @returns {Undefined} - does not return a value
+   */
+  function uncaughtExceptionHandler(err) {
+    var em = new ErrorMessage();
+
+    errorHandlerRouter(err, em);
+    client.sendError(em, handleProcessExit);
+  }
 
   if (config.reportUncaughtExceptions === false) {
     // Do not attach a listener to the process
     return null;
   }
 
-  return attachUncaughtExceptionHandler(client, config);
+  return process.on('uncaughtException', uncaughtExceptionHandler);
```

Comment:
`process` object is an `EventEmitter` instance. Is your question about why we are returning anything at all? It is a common pattern to chain `EventEmitter.on` calls. From that point of view, it makes sense to return the `EventEmitter` (i.e. `process`).